### PR TITLE
Bug 1800700 - Add default search suggest header

### PR DESCRIPTION
**Pull request by sarah541**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### GitHub Automation

Used by GitHub Actions.

### CODE REVIEW

File: android-components/components/feature/awesomebar/src/main/java/mozilla/components/feature/awesomebar/provider/SearchSuggestionProvider.kt [hunk 1]
```diff
@@ -99,6 +100,7 @@ class SearchSuggestionProvider private constructor(
      * @param filterExactMatch If true filters out suggestions that exactly match the entered text.
      * @param private When set to `true` then all requests to search engines will be made in private
      * mode.
+     *  @param suggestionsHeader optional parameter to specify if the suggestion should have a header
```
Comment (gabrielluong) [inline android-components/components/feature/awesomebar/src/main/java/mozilla/components/feature/awesomebar/provider/SearchSuggestionProvider.kt:103]:
```suggestion
     * @param suggestionsHeader optional parameter to specify if the suggestion should have a header
```
Just a drive-by

File: android-components/components/compose/awesomebar/src/main/java/mozilla/components/compose/browser/awesomebar/internal/Suggestions.kt [hunk 2]
```diff
@@ -37,11 +37,9 @@ internal fun Suggestions(
         modifier = Modifier.testTag("mozac.awesomebar.suggestions"),
     ) {
         suggestions.forEach { (group, suggestions) ->
-            if (suggestions.isNotEmpty()) {
-                group.title?.let { title ->
-                    item(group.id) {
-                        SuggestionGroup(title, colors)
-                    }
+            if (suggestions.isNotEmpty() && !group.title.isNullOrEmpty()) {
```
Comment (rocketsroger) [inline android-components/components/compose/awesomebar/src/main/java/mozilla/components/compose/browser/awesomebar/internal/Suggestions.kt:40]: To avoid using `!!` maybe this is better?
```suggestion
            val title = group.title
            if (suggestions.isNotEmpty() && !title.isNullOrEmpty()) {
                item(group.id) {
                    SuggestionGroup(title, colors)
                }
            }
```

File: android-components/components/feature/awesomebar/src/main/java/mozilla/components/feature/awesomebar/provider/SearchSuggestionProvider.kt [hunk 3]
```diff
@@ -99,6 +100,7 @@ class SearchSuggestionProvider private constructor(
      * @param filterExactMatch If true filters out suggestions that exactly match the entered text.
      * @param private When set to `true` then all requests to search engines will be made in private
      * mode.
+     * @param suggestionsHeader optional parameter to specify if the suggestion should have a header
```
Comment (gabrielluong) [inline android-components/components/feature/awesomebar/src/main/java/mozilla/components/feature/awesomebar/provider/SearchSuggestionProvider.kt:103]:
```suggestion
     * @param suggestionsHeader Optional suggestions header to display.
```

File: android-components/components/compose/awesomebar/src/main/java/mozilla/components/compose/browser/awesomebar/internal/Suggestions.kt (hunk 2 above)
Comment (gabrielluong) [reply android-components/components/compose/awesomebar/src/main/java/mozilla/components/compose/browser/awesomebar/internal/Suggestions.kt:40]: Agreed, we should try to avoid `!!`

**Comment by gabrielluong** [comment]
Roger and I requested tests and an entry in the changelog

**Comment by sarah541** [comment]
> Roger and I requested tests and an entry in the changelog

No good tests to add. changelog updated.

**Review by rocketsroger — APPROVED** [review]

**Comment by rocketsroger** [comment]
@Mergifyio backport releases_v108

**Comment by rocketsroger** [comment]
Uplift for releases_v108 https://github.com/mozilla-mobile/firefox-android/pull/205
